### PR TITLE
switch a symbol_table to use an unordered_map and move it to interpreter

### PR DESCRIPTION
**Pull request by Krovatkin**

### CODE REVIEW

File: torch/csrc/jit/runtime/interpreter.cpp [hunk 1]
```diff
@@ -900,6 +900,69 @@ struct CodeImpl {
...
+    if (s.is_static()) {
+      return s.static_size();
+    }
+    return data_[s];
+  }
+  void assign(c10::ShapeSymbol s, Dimension v) {
+    TORCH_INTERNAL_ASSERT(!s.is_static());
+    data_[s] = v;
+  }
+  std::unordered_map<c10::ShapeSymbol, Dimension> data_;
```
Comment (zdevito) [inline torch/csrc/jit/runtime/interpreter.cpp:929]: should be private

File: torch/csrc/jit/runtime/interpreter.cpp [hunk 2]
```diff
@@ -900,6 +900,69 @@ struct CodeImpl {
...
+  // Tries to assign dimension values from `new_sizes` to
+  // `ShapeSymbol`s `sym_shapes`.
+  // Returns `true` if every dimension value from `new_sizes`
+  // can be assigned to the corresponding `ShapeSymbol` from
+  // `sym_shapes`
+  // A dimension value can be assigned to a `ShapeSymbol`
+  // * if the symbol isn't assigned yet any dimension value
+  // * if the symbol is assigned and its value is equal to
+  // the dimension value from `new_sizes`
+  bool ShapeSymbolTable::bindSymbolicShapes(
```
Comment (zdevito) [inline torch/csrc/jit/runtime/interpreter.cpp:939]: ShapeSymbolTable:: ?

**Review by zdevito — APPROVED** [review]

File: torch/csrc/jit/runtime/interpreter.cpp (hunk 1 above)
Comment (Krovatkin) [reply torch/csrc/jit/runtime/interpreter.cpp:929]: will fix.

File: torch/csrc/jit/runtime/interpreter.cpp (hunk 2 above)
Comment (Krovatkin) [reply torch/csrc/jit/runtime/interpreter.cpp]: will fix

**Pull request closed without merging**